components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -70,6 +71,22 @@ pub fn day_of_year(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option
     Ok(Some(Int::from(day)))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn week_of_year(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>> {
+    if t.is_none() {
+        return Ok(None);
+    }
+    let t = t.as_ref().unwrap();
+    if t.invalid_zero() {
```

Comment:
Hi, the original implementation use `t.is_zero()`. Could you explain why do you want to change it? Also @iosmanthus PTAL about this change.

Reply:
Hello, it seems to me `is_zero` means all components of a time variable (i.e. year, month, date and others) equal to zero, while `invalid_zero` means that either its month or its date is zero. Maybe we should use `invalid_zero` to see whether the input is incorrect? Actually the original implementation using `is_zero` confuses me.

Reply:
It seems that the code is ported from TiDB.

https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L1453

zero DateTime is invalid in this case, and maybe invalid_zero will never occurred here?

Reply:
And you can try it in mysql, `SELECT WEEKOFYEAR("0-0-0");`

A warning `Incorrect datetime value: '0-0-0'` is generated.

Reply:
@iosmanthus PTAL

Reply:
It should be `t.invalid_zero()`, because `0000-00-00` is `invalid-zero` and so does `2019-05-00` or `2019-00-01`. All of them should report `NULL` when they are processed by `week_of_year`

Reply:
So it seems that we need to update many time related function, both at TiDB and TiKV